### ISSUE TITLE
Incorrect printout

### ISSUE THREAD

**Issue by Mark3032** [thunderbird/thunderbird-android#8701]
### Checklist

- [x] I have used the search function to see if someone else has already submitted the same bug report.
- [x] I will describe the problem with as much detail as possible.

### App

Thunderbird for Android

### App version

8.2

### Where did you get the app from?

Google Play

### Android version

14

### Device model

Samsung Galaxy A52

### Steps to reproduce

First off, I am just a regular user and will try to describe the problem as best as I can. I am having an issue with printing emails. I didn’t have this problem with K9 back in the summer, or at least I didn’t notice it when I switched to K9. I suspect it started happening after an update to K9.

That’s why I installed Thunderbird Mail on my Android phone today, but unfortunately, the exact same issue occurs. Instead of printing the email with the text, if the email contains a URL like https://www.thunderbird-mail.de, the landing page of that URL gets printed. For example, instead of printing the order confirmation from Apotheke.at, the homepage of www.apotheke.at is printed because the order confirmation email naturally includes the URL http://www.apotheke.at somewhere.

It doesn’t matter whether I use Samsung or Brother as the printer service (I have multifunction devices from both brands in my office).

Printing only works without issues when the document or email does not contain an embedded web address. If such an address is present, the text of the document isn’t printed; instead, the landing page of the web address is printed. For instance, the order confirmation from Amazon isn’t printed; instead, the homepage (landing page) of http://www.amazon.com is printed.

This issue is not just affecting me but also other users – there is a dedicated thread about it in the Thunderbird forum: https://www.thunderbird-mail.de/forum/thread/95888-thunderbird-android-8-1-keine-m%C3%B6glichkeit-zu-drucken/?postID=550173#post550173.

### Expected behavior

-

### Actual behavior

-

### Logs

-

**Comment by dg1rue** [comment]
I use Google pixel 7. No additional App installed for printer. I need to print out order reply messages. Please  add print option ASAP.

**Comment by kewisch** [comment]
We're going to track the feature request on printing here, please upvote if you care about printing: https://connect.mozilla.org/t5/ideas/ability-to-print-email-messages/idi-p/71885

**Comment by systemoperator** [comment]
I can also confirm this issue on any phone with Thunderbird Android. Because, native printing is still not integrated, there is no other way than sharing the mail with e.g. a printer app. But when the mail is shared and contains any HTML URL, instead of printing the mail as expected, the first URL in the mail is loaded and the website's content is printed, which renders printing impossible.

@kewisch I would not consider this as a feature request, because it's an issue concerning the sharing functionality. Besides the issue with the sharing functionality, the proper Android printing API should be integrated.

**Comment by kewisch** [comment]
Whether this is considered a feature request or a bug doesn’t make a big difference right now, as it’s not something my team is actively pursuing. Since we’re not using the printing API at the moment, framing it as a feature request feels like the better fit. Anyone interested in taking this on is more than welcome to contribute.

I agree it isn't a great experience right now, I hope there will be time for this in the future!

**Comment by systemoperator** [comment]
@kewisch In my opinion, it's confusing to mark a clearly unresolved issue as "complete" simply because the team is not actively pursuing it and no functional workaround exists. I think, issues should remain open until they are either resolved or a reliably working alternative is provided. Closing them prematurely risks losing track of valid problems, leading to repeated reports and duplicate issues later on.

**Comment by kewisch** [comment]
Yes, GitHub is confusing in that way. I only have "complete" and "not planned", and I think not planned sends an even worse message. We're going to get duplicates either way, but keeping it open in two places makes it harder to track on our end. We've decided to use Mozilla Connect for enhancements, and that this is primarily an enhancement. I know this isn't the answer you might have wished for and acknowledge that you see it differently.
